### PR TITLE
[OSCD] Local System Accounts Discovery

### PR DESCRIPTION
**Pull request by alejandroortuno**
Covering:

Linux Task 19  T1087.001: Local Account #1011
MacOs Task 34 T1136.001: Local Account #1012

Regarding the outcome of #1048 we can consider renaming the file name prefix and also move it into the right folder.

### CODE REVIEW

File: rules/linux/lnx_local_account.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+title: Local System Accounts Discovery
+id: b45e3d6f-42c6-47d8-a478-df6bd6cf534c
+status: experimental
+description: Detects enumeration of local systeam accounts
+author: Alejandro Ortuno, oscd.community
+date: 2020/10/08
+references:
+  - https://attack.mitre.org/techniques/T1087/001/
```
Comment (yugoslavskiy) [inline rules/linux/lnx_local_account.yml:8]: Hello @alejandroortuno! Could you please remove reference to the ATT&CK as it already mentioned in the `tags` filed?

File: rules/linux/macos_local_account.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+title: Local System Accounts Discovery
+id: ddf36b67-e872-4507-ab2e-46bda21b842c
+status: experimental
+description: Detects enumeration of local systeam accounts on MacOS
+author: Alejandro Ortuno, oscd.community
+date: 2020/10/08
+references:
+  - https://attack.mitre.org/techniques/T1087/001/
```
Comment (yugoslavskiy) [inline rules/linux/macos_local_account.yml:8]: could you please do the same here?

File: rules/linux/macos_local_account.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1087.001/T1087.001.md
+logsource:
+  product: macos
+detection:
+  selection:
+    CommandLine|contains:
+      - 'dscl . list /Users'
+      - 'dscacheutil -q user'
```
Comment (yugoslavskiy) [inline rules/linux/macos_local_account.yml:16]: It make sense to split `processname` and `commandline`. It is cheaper from resources point of view (more efficient). 

Here is how santa log looks like for the `dscl . list /Users` command:

```
"process": {
      "args": [
        "dscl",
        ".",
        "list",
        "/Users"
      ],
      "start": "2020-10-10T04:07:08.961Z",
      "pid": 23138,
      "executable": "/usr/bin/dscl",
      "ppid": 21813
    },
```

Here is how santa log looks like for the `dscacheutil -q user` command:

```
"process": {
      "args": [
        "dscacheutil",
        "-q",
        "user"
      ],
      "start": "2020-10-10T04:07:15.514Z",
      "pid": 23155,
      "executable": "/usr/bin/dscacheutil",
      "ppid": 21813
    },
```

File: rules/linux/unix_local_account.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+title: Local System Accounts Discovery
+id: 396fe688-65d9-4828-a078-ed17551f9a8a
+status: experimental
+description: Detects enumeration of local systeam accounts
+author: Alejandro Ortuno, oscd.community
+date: 2020/10/08
+references:
+  - https://attack.mitre.org/techniques/T1087/001/
```
Comment (yugoslavskiy) [inline rules/linux/unix_local_account.yml:8]: could you please do the same here?

File: rules/linux/unix_local_account.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+logsource:
+  product: unix
+detection:
+  selection:
+    CommandLine|contains:
+      - 'cat /etc/passwd'
+      - 'cat /etc/sudoers'
+      - 'id '
+      - "'x:0:'"
+      - 'lsof -u'
```
Comment (yugoslavskiy) [inline rules/linux/unix_local_account.yml:19]: could you please split this rule for Linux and macOS, and apply the same approach splitting `processname` and `commandline`? 

thank you!

**Comment by alejandroortuno** [comment]
@yugoslavskiy  let me know if the latest changes make sense with the split between `processname` and `commandline` and the move from the unix file into the individuals liux / macos.

File: rules/linux/macos_local_account.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+id: ddf36b67-e872-4507-ab2e-46bda21b842c
+status: experimental
+description: Detects enumeration of local systeam accounts on MacOS
+author: Alejandro Ortuno, oscd.community
+date: 2020/10/08
+references:
+  - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1087.001/T1087.001.md
+logsource:
+  category: process_creation
+  product: macos
```
Comment (alejandroortuno) [inline rules/linux/macos_local_account.yml:11]: @yugoslavskiy this is something I didnt have 100% sure that we can reference a product named `macos`? I could not find any reference on within the /tools/config folder

File: rules/linux/lnx_local_account.yml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+date: 2020/10/08
+references:
+  - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1087.001/T1087.001.md
+logsource:
+  category: process_creation
+  product: linux
+detection:
+  selection_1:
+    ProcessName|contains:
+      - '*/lastlog'
```
Comment (yugoslavskiy) [inline rules/linux/lnx_local_account.yml:15]: Hello @alejandroortuno ! could you please apply `endswith` [Value Modifier](https://github.com/Neo23x0/sigma/wiki/Rule-Creation-Guide#value-modifiers) here?

File: rules/linux/lnx_local_account.yml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  selection_2:
+    CommandLine|contains:
+      - "'x:0:'"
+  selection_3:
+    ProcessName|contains:
+      - '*/cat'
```
Comment (yugoslavskiy) [inline rules/linux/lnx_local_account.yml:21]: Could you please apply `endswith` [Value Modifier](https://github.com/Neo23x0/sigma/wiki/Rule-Creation-Guide#value-modifiers) here as well?

File: rules/linux/lnx_local_account.yml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    CommandLine|contains:
+      - '/etc/passwd'
+      - '/etc/sudoers'
+  selection_4:
+    ProcessName|contains:
+      - '*/id'
```
Comment (yugoslavskiy) [inline rules/linux/lnx_local_account.yml:27]: Could you please apply `endswith` [Value Modifier](https://github.com/Neo23x0/sigma/wiki/Rule-Creation-Guide#value-modifiers) here as well?

File: rules/linux/lnx_local_account.yml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  selection_5:
+    ProcessName|contains:
+      - '*/lsof'
```
Comment (yugoslavskiy) [inline rules/linux/lnx_local_account.yml:30]: Could you please apply `endswith` [Value Modifier](https://github.com/Neo23x0/sigma/wiki/Rule-Creation-Guide#value-modifiers) here as well?

File: rules/linux/macos_local_account.yml [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+detection:
+  selection_1:
+    ProcessName|contains:
+      - '*/dscl'
```
Comment (yugoslavskiy) [inline rules/linux/macos_local_account.yml:15]: Could you please apply `endswith` [Value Modifier](https://github.com/Neo23x0/sigma/wiki/Rule-Creation-Guide#value-modifiers) here as well?

File: rules/linux/macos_local_account.yml [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    CommandLine|contains:
+      - '. list /users'
+  selection_2:
+    ProcessName|contains:
+      - '*/dscacheutil'
```
Comment (yugoslavskiy) [inline rules/linux/macos_local_account.yml:20]: Could you please apply `endswith` [Value Modifier](https://github.com/Neo23x0/sigma/wiki/Rule-Creation-Guide#value-modifiers) here as well?

File: rules/linux/macos_local_account.yml [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+      - "'x:0:'"
+  selection_4:
+    ProcessName|contains:
+      - '*/cat'
+    CommandLine|contains:
+      - '/etc/passwd'
+      - '/etc/sudoers'
+  selection_5:
+    ProcessName|contains:
+      - '*/id'
```
Comment (yugoslavskiy) [inline rules/linux/macos_local_account.yml:34]: And here (:

File: rules/linux/macos_local_account.yml [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+  selection_6:
+    ProcessName|contains:
+      - '*/lsof'
```
Comment (yugoslavskiy) [inline rules/linux/macos_local_account.yml:37]: And also here (:

File: rules/linux/macos_local_account.yml (hunk 6 above)
Comment (yugoslavskiy) [reply rules/linux/macos_local_account.yml:11]: You are right, there is no such service yet, but the only reason for it is that there was no macos rules (: so you are kinda inventing it right now. I believe the converter will be updated shortly. Thank you!

File: rules/linux/macos_local_account.yml [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    ProcessName|contains:
+      - '*/dscacheutil'
+    CommandLine|contains:
+      - '-q user'
+  selection_3:
+    CommandLine|contains:
+      - "'x:0:'"
+  selection_4:
+    ProcessName|contains:
+      - '*/cat'
```
Comment (yugoslavskiy) [inline rules/linux/macos_local_account.yml:28]: And here (:

**Comment by alejandroortuno** [comment]
cool stuff @yugoslavskiy, that was my first sigma rule so appreciate your effort into reviewing it. I will take this learning on the other rules I am building. thanks :)

File: rules/linux/lnx_local_account.yml [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+date: 2020/10/08
+references:
+  - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1087.001/T1087.001.md
+logsource:
+  category: process_creation
+  product: linux
+detection:
+  selection_1:
+    ProcessName|endswith:
+      - '*/lastlog'
```
Comment (yugoslavskiy) [inline rules/linux/lnx_local_account.yml:15]: Hello @alejandroortuno ! Could you please remove `'*'` from the `'*/lastlog'` as it's already supposed by the `endswith`?

File: rules/linux/lnx_local_account.yml [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+  selection_2:
+    CommandLine|contains:
+      - "'x:0:'"
+  selection_3:
+    ProcessName|endswith:
+      - '*/cat'
```
Comment (yugoslavskiy) [inline rules/linux/lnx_local_account.yml:21]: Could you please remove `'*'` from the `'*/cat'` as it's already supposed by the `endswith`?

File: rules/linux/lnx_local_account.yml [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+    CommandLine|contains:
+      - '/etc/passwd'
+      - '/etc/sudoers'
+  selection_4:
+    ProcessName|endswith:
+      - '*/id'
```
Comment (yugoslavskiy) [inline rules/linux/lnx_local_account.yml:27]: Could you please remove `'*'` from the `'*/id'` as it's already supposed by the `endswith`? (:

File: rules/linux/lnx_local_account.yml [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+  selection_5:
+    ProcessName|endswith:
+      - '*/lsof'
```
Comment (yugoslavskiy) [inline rules/linux/lnx_local_account.yml:30]: And here as well (:

File: rules/linux/macos_local_account.yml (hunk 11 above)
Comment (yugoslavskiy) [reply rules/linux/macos_local_account.yml:15]: Could you please do the same here? (:

Comment (yugoslavskiy) [reply rules/linux/macos_local_account.yml:15]: And apply it to all of your rules?

File: rules/linux/macos_local_account.yml [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+  - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1087.001/T1087.001.md
+logsource:
+  category: process_creation
+  product: macos
+detection:
+  selection_1:
+    ProcessName|endswith:
+      - '*/dscl'
+    CommandLine|contains:
+      - '. list /users'
```
Comment (yugoslavskiy) [inline rules/linux/macos_local_account.yml:17]: I would rather split it to the following:

```
    CommandLine|contains|all:
      - 'list'
      - '/users'
```

That would make it more stable and hard to bypass. In the current state you could bypass it just adding an extra space (:

File: rules/linux/macos_local_account.yml [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+  selection_2:
+    ProcessName|endswith:
+      - '*/dscacheutil'
+    CommandLine|contains:
+      - '-q user'
```
Comment (yugoslavskiy) [inline rules/linux/macos_local_account.yml:22]: and the same here:

```
 CommandLine|contains|all:
      - '-q'
      - 'user'
```

File: rules/linux/macos_local_account.yml (hunk 20 above)
Comment (alejandroortuno) [reply rules/linux/macos_local_account.yml:17]: 100% agree!

**Comment by yugoslavskiy** [comment]
Thank you very much, @alejandroortuno! Well done!

**Review by yugoslavskiy — APPROVED** [review]
@thomaspatzke @Neo23x0 guys, I believe that this PR could be merged (:
please let us know what do you think about it if there is anything you would like to change.

**Pull request merged**